src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -479,41 +520,37 @@ func (sm *StorageMinerActor_I) verifySeal(onChainInfo sector.OnChainSealVerifyIn
 
 	new(proving.StorageProvingSubsystem_I).VerifySeal(&sector.SealVerifyInfo_I{
 		SectorID_: &sector.SectorID_I{
-			MinerID_: sm.Info().Worker(), // TODO: This is actually miner address. MinerID needs to be derived.
+			MinerID_: st.Info().Worker(), // TODO: This is actually miner address. MinerID needs to be derived.
 			Number_:  onChainInfo.SectorNumber(),
 		},
 
 		OnChain_:    onChainInfo,
 		PieceInfos_: pieceInfos,
 	})
-	return true
+	return false // TODO: finish
 }
 
-func (sm *StorageMinerActor_I) checkIfSectorExists(sectorNo sector.SectorNumber) bool {
-	_, sectorExists := sm.Sectors()[sectorNo]
-	if sectorExists {
-		return true
-	}
-	return false
+func (st *StorageMinerActorState_I) _sectorExists(rt Runtime, sectorNo sector.SectorNumber) bool {
+	_, found := st.Sectors()[sectorNo]
+	return found
 }
 
 // Currently deals must be posted on chain via sma.PublishStorageDeals before CommitSector
 // TODO(optimization): CommitSector could contain a list of deals that are not published yet.
-func (sm *StorageMinerActor_I) CommitSector(onChainInfo sector.OnChainSealVerifyInfo) {
+func (a *StorageMinerActorCode_I) CommitSector(rt Runtime, onChainInfo sector.OnChainSealVerifyInfo) {
+	h, st := a.State(rt)
 
-	isSealVerified := sm.verifySeal(onChainInfo)
-	if !isSealVerified {
-		// TODO: proper failure
-		panic("Seal is not verified")
+	isSealVerificationCorrect := st._isSealVerificationCorrect(rt, onChainInfo)
+	if !isSealVerificationCorrect {
+		rt.Fatal("Seal verification failed")
```

Comment:
let's flag errors as something to discuss later

Reply:
OK -- adding comments in those locations.